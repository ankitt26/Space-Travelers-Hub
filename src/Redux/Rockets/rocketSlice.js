import { createSlice } from "@reduxjs/toolkit";
const rocketSlice = createSlice({
  name: "rockets",
  initialState: [],
  reducers: {
    addRocket: () => {},
    removeRocket: () => {},
  },
});

export default rocketSlice.reducer;
