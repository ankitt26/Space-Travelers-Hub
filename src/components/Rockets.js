import { useSelector } from 'react-redux';
import styles from '../CSS/Rockets.module.css';
import RocketButton from './RocketButton';

const Rockets = () => {
  const data = useSelector((state) => state.Rockets.rocketData);
  const display = data.map((rocket) => (
    <main className={styles.rocket} key={rocket.id}>
      <img className={styles.image} src={rocket.flickr_images} alt="space" />
      <div className={styles.details}>
        <h2 className={styles.name}>{rocket.name}</h2>
        <p className={styles.description}>{rocket.description}</p>
        <RocketButton />
      </div>
    </main>
  ));
  return <section className={styles.rocket_page}>{display}</section>;
};
export default Rockets;
