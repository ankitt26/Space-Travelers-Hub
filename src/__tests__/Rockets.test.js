import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router';
import Rockets from '../components/Rockets';
import Store from '../Redux/store';

it('MyProfile component test', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <Provider store={Store}>
          <Rockets />
        </Provider>
      </MemoryRouter>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
