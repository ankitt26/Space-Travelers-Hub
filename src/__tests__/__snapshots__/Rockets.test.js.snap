// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyProfile component test 1`] = `
<section
  className="rocket_page"
/>
`;
